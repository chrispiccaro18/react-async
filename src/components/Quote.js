import React from 'react';
import PropTypes from 'prop-types';
import { normalizeCharacter } from '../services/futuramaApi';

function Quote({ quote, character, image }) {
  const normalCharName = normalizeCharacter(character);
  return (
    <li>
      <img src={image} alt={`image of ${character}`} />
      <h3>{normalCharName}</h3>
      <p>{quote}</p>
    </li>
  );
}

Quote.propTypes = {
  quote: PropTypes.string.isRequired,
  character: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired
};

export default Quote;
