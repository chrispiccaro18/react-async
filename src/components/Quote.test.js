import React from 'react';
import { shallow } from 'enzyme';
import Quote from './Quote';

describe('Quote component', () => {
  it('renders Quote component', () => {
    const props = {
      quote: 'Problem solved. You two fight to the death and I\'ll cook the loser.',
      character: 'Bender',
      image: 'https://res.cloudinary.com/dzxqhkyqd/image/fetch/c_scale,w_500/https://res.cloudinary.com/dzxqhkyqd/image/upload/v1552429540/bender.png'
    };
    const wrapper = shallow(<Quote {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
