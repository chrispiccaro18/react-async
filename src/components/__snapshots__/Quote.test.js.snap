// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote component renders Quote component 1`] = `
<li>
  <img
    alt="image of Bender"
    src="https://res.cloudinary.com/dzxqhkyqd/image/fetch/c_scale,w_500/https://res.cloudinary.com/dzxqhkyqd/image/upload/v1552429540/bender.png"
  />
  <h3>
    Bender
  </h3>
  <p>
    Problem solved. You two fight to the death and I'll cook the loser.
  </p>
</li>
`;
